File: KadanesAlgorithm.java
class KadanesAlgorithm{
  public static int kadanesAlgorithm(int[] array) {
   // Write your code here.
   int maxSum =  array[0];
   int previousMax = array[0];
   for ( int i = 1 ; i < array.length ; i++ ){
     previousMax = array[i] > (array[i]+previousMax) ? array[i]: (array[i]+previousMax);
     maxSum = previousMax > maxSum ? previousMax : maxSum;
   }
   return maxSum;
 }
}

File: KadanesAlgorithm.cpp
#include <vector>
using namespace std;
int kadanesAlgorithm(vector<int> array) {
  // Write your code here.
	int tempMax = array[0];
	int previous = array[0];
	for(int i = 1; i < array.size(); i++){
		previous = max(array[i], array[i]+ previous);
		tempMax = max(previous, tempMax);
	}
  return tempMax;
}
